contracts/Staking.sol

```diff
@@ -192,24 +187,26 @@ contract Staking is IERC20, IERC20Metadata {
         require(token.transferFrom(msg.sender,address(this),_amount),"");
 
         UserInfo storage user = userInfo[_to];
-        user.accamulatedReward += accamulatedRewardPerShare * user.amount - user.rewardDebt;
+        user.accumulatedReward += calculateRewardAmount(user.amount) - user.rewardDebt;
         totalAmount += _amount;
         user.amount += _amount;
-        user.rewardDebt = accamulatedRewardPerShare * user.amount;
+        user.rewardDebt = calculateRewardAmount(user.amount);
     }
 
     function harvest(uint256 _amount) public whenNotPaused {
         updateRewardPool();
         UserInfo storage user = userInfo[msg.sender];
         // +1 to prevent efforts in scum of tstamp
-        require(user.lastHarvestTimestamp + harvestInterval + 1 > block.timestamp || 
-            user.lastHarvestTimestamp == 0, "not so fast");
+        require(user.lastHarvestTimestamp + harvestInterval + 1 <= block.timestamp || 
+            user.lastHarvestTimestamp == 0, "Staking: less than 24 hours since last harvest");
         user.lastHarvestTimestamp = block.timestamp;
 
-        user.accamulatedReward += accamulatedRewardPerShare * user.amount - user.rewardDebt;
+        user.accumulatedReward += calculateRewardAmount(user.amount) - user.rewardDebt;
+        user.rewardDebt = calculateRewardAmount(user.amount);
```

Comment:
Aren't we spending extra gas here by calling calculateRewardAmount two times?

Reply:
Good notice! Will fix that